AEPMessaging/Sources/MessagingPropositionItem.swift

```diff
@@ -70,6 +70,45 @@ public class MessagingPropositionItem: NSObject, Codable {
 }
 
 public extension MessagingPropositionItem {
+    /// Tracks interaction with the given proposition item.
+    ///
+    /// - Parameter eventType: an enum specifying event type for the interaction.
+    func track(eventType: MessagingEdgeEventType) {
+        guard let propositionInteractionXdm = generateInteractionXdm(forEventType: eventType) else {
+            Log.debug(label: MessagingConstants.LOG_TAG,
+                      "Cannot track proposition interaction for item \(itemId), could not generate interactions XDM.")
+            return
+        }
+
+        let eventData: [String: Any] = [
+            MessagingConstants.Event.Data.Key.TRACK_PROPOSITIONS: true,
+            MessagingConstants.Event.Data.Key.PROPOSITION_INTERACTION: propositionInteractionXdm
+        ]
+
+        let event = Event(name: MessagingConstants.Event.Name.TRACK_PROPOSITIONS,
+                          type: EventType.messaging,
+                          source: EventSource.requestContent,
+                          data: eventData)
+
+        MobileCore.dispatch(event: event)
+    }
+
+    /// Creates a dictionary containing XDM data for interaction with the given proposition item, for the provided event type.
+    ///
+    /// If the proposition reference within the item is released and no longer valid, the method returns `nil`.
+    ///
+    /// - Parameter eventType: an enum specifying event type for the interaction.
+    /// - Returns A dictionary containing XDM data for the propositon interaction.
+    func generateInteractionXdm(forEventType eventType: MessagingEdgeEventType) -> [String: Any]? {
+        guard let proposition = proposition else {
+            Log.debug(label: MessagingConstants.LOG_TAG,
+                      "Cannot generate interaction XDM for item \(itemId), proposition reference is not available.")
+            return nil
+        }
+
+        return MessagingPropositionInteraction(eventType: eventType, interaction: nil, propositionInfo: PropositionInfo.fromProposition(proposition), itemId: itemId).xdm
```

Comment:
do we never need to pass a value for the interaction?  `interaction: nil`

Reply:
Yes, as per the latest item level tracking spec we will allow customers to pass interaction as a custom data attribute (data-*). The API method on MessagingPropositionItem will accept additional parameters for `item tokens` and `interaction`.

Reply:
gotcha, so coming in a future update then?

Reply:
yup, that's why I believe we should add that support before GA. Otherwise we will need to expose another API method after GA for additional tracking info.